cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
```

Comment:
I’m wondering how much longer it takes to repeatedly apply a regexp as opposed to using the `strings` package. Could you do a micro-benchmark of the new code against the old code? Given that this piece of code sits in the critical path for each and every query, I’d hate to introduce a performance regression here :).